tb/cocotb/soc/periph_fw/gen_periph_hex.py

```
@@ -0,0 +1,420 @@
+  x31 = result marker (1=pass, -1=fail)
+"""
+
+import os
+import sys
+
+# Add project root to path so we can import riscv_encoder
+_PROJ_ROOT = os.path.abspath(os.path.join(os.path.dirname(__file__), '..', '..', '..', '..'))
+if _PROJ_ROOT not in sys.path:
+    sys.path.insert(0, _PROJ_ROOT)
+
+from sim.riscv_encoder import (
```

Flake8 (7.3.0)

[error] 'sim.riscv_encoder.ORI' imported but unused

(F401)